azure-pipelines.yml

```diff
@@ -1,95 +1,183 @@
 ---
 trigger:
-  - none
+  - master
 
 parameters:
   - name: environment
-    displayName: Environment
     type: string
-    default: 'sbox'
-    values:
-      - sbox
-      - test
+    default: 'sbox,test'
 
 variables:
   - group: HMI-APIM-Common
-  - ${{ if eq(parameters.environment, 'sbox') }}:
-    - group: HMI-APIM-BUILD-SBOX
-  - ${{ if eq(parameters.environment, 'test') }}:
-    - group: HMI-APIM-BUILD-TEST
 
 stages:
 
-  - stage: CI
-    displayName: Validate the code
+  - stage: SANDBOX
+    displayName: Build and Test SANDBOX APIM
+    
     jobs:
-    - job: validatetf
-      pool:
-        vmImage: 'ubuntu-latest'
-      steps:
-      - template: pipeline/steps/tf-test.yaml
-
-  - stage: BUILD
-    displayName: Build infrastructure with Terraform
-    jobs:
-      - job: terraform
+    - ${{ if contains(parameters.environment, 'sbox') }}:
+
+      - job: ValidateTF
+        displayName: Validate the code
+        pool:
+          vmImage: 'ubuntu-latest'
+        steps:
+        - template: pipeline/steps/tf-test.yaml
+
+      - job: Terraform
+        variables:
+        - group: HMI-APIM-BUILD-SBOX
+        displayName: Build infrastructure with Terraform
         pool:
           vmImage: 'ubuntu-latest'
         steps:
         - template: pipeline/steps/tf-install.yaml
         - template: pipeline/steps/tf-init.yaml
           parameters:
-            resourceGroup: $(devopsResourceGroup)
-            storageAccount: $(devopsStorageAccount)
+            subscription: 'DTS-SHAREDSERVICES-SBOX'
         - template: pipeline/steps/tf-plan.yaml
+          parameters:
+            subscription: 'DTS-SHAREDSERVICES-SBOX'
         - template: pipeline/steps/tf-apply.yaml
+          parameters:
+            subscription: 'DTS-SHAREDSERVICES-SBOX'
         - template: pipeline/steps/export-keyvault.yaml
           parameters:
+            subscription: 'DTS-SHAREDSERVICES-SBOX'
```

Comment:
define a sub object or variable group and read from there rather than defining this string over and over.